sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -836,15 +848,22 @@ pi_result _pi_ze_event_list_t::createAndRetainPiZeEventList(
         this->PiEventList[TmpListLength] = EventList[I];
         TmpListLength += 1;
       }
-
-      this->Length = TmpListLength;
-    } catch (...) {
-      return PI_OUT_OF_HOST_MEMORY;
     }
 
-    for (pi_uint32 I = 0; I < this->Length; I++) {
-      PI_CALL(piEventRetain(this->PiEventList[I]));
+    if (CurQueue->isInOrderQueue()) {
```

Comment:
Please add a comment explaining why we do this

Reply:
done.